knowledge-base/adr/0028-public-api-extraction.md

```diff
@@ -27,8 +27,12 @@ This was convenient for us, but we had no distinction between:
 
 Our new strategy should provide the following options:
 
-- `Exported Public API`: This is a cautious decision and consumer can rely on stable contract for minor versions. This API is reachable via the root level import.
-- `Exported Internal API`: Exported for technical reasons but not meant to be used by consumer so no stable contract. This API is not accessible without [dirty tricks](https://www.goodreads.com/quotes/7714510-c-protects-against-accident-not-against-fraud).
+- `Exported Public API`: This is a cautious decision and consumer can rely on stable contract for minor versions.
+  It contains only objects where we see (re)use potential for customers.
+  This API is reachable via the root level import e.g. `@sap-cloud-sdk/odata-v2`.
+- `Exported Internal API`: Exported for technical reasons but not meant to be used by consumer.
+  We keep this API stable for path versions.
```

Comment:
Do we really want to have this contract?
I would not make any promises if possible and only tells please use internal APIs with caution.

Reply:
Yes I would like to keep for patch releases the internal API stable. I mean we can always do a minor release if needed.